### DISCUSSION TITLE
Secrets managed by a third party

### DISCUSSION THREAD

**Discussion by kuratowskiy** [fluxcd/flux2#1251] [Q&A]
I have a general question about secrets that are managed by a third party. If I have a dependent manifest to a secret that is not tracked by any of my repositories, and the secret changes, will the cluster be reconciled with the change?

I have a container that depends on the existence of a secret that is managed by cert-manager. My chart includes the Certificate manifest, and the manifest that describes the deployment. When Flux applies the HelmRelease, the Certificate is not immediately issued, and the number of reconciliation retries is exhausted, for the reason that the secret does not yet exist. What is the correct way to approach this problem?

**Comment by Legion2** [comment]
I also use cert-manager and manage Certificates with HelmRelease and it works. Flux should not be aware of the secret, because the secret is managed by cert-manager and used by the deployment.

**Comment by kuratowskiy** [comment]
Yeah because this is not GitOps but still gay I recommend it

**Comment by kuratowskiy** [comment]
I didn't know that secrets changes are remounted to the pod. I also didn't know that secrets subpath mounted are not remounted because I cannot read documentation. Thanks for keeping the forum newbie friendly.
